src/platform/Zephyr/WiFiManager.cpp

```diff
@@ -355,5 +354,102 @@ void WiFiManager::PollTimerCallback()
     }
 }
 
+CHIP_ERROR WiFiManager::GetWiFiInfo(WiFiInfo & info) const
+{
+    VerifyOrReturnError(nullptr != wpa_s_0, CHIP_ERROR_INTERNAL);
+
+    static uint8_t sBssid[ETH_ALEN];
+    if (WiFiManager::StationStatus::CONNECTED <= GetStationStatus())
+    {
+        memcpy(sBssid, wpa_s_0->bssid, ETH_ALEN);
+        info.mBssId        = ByteSpan(sBssid, ETH_ALEN);
+        info.mSecurityType = GetSecurityType();
+        // TODO: this should reflect the real connection compliance
+        // i.e. the AP might support WiFi 5 only even though the station
+        // is WiFi 6 ready (so the connection is WiFi 5 effectively).
+        // For now just return what the station supports.
+        info.mWiFiVersion = EMBER_ZCL_WI_FI_VERSION_TYPE_802__11AX;
+
+        wpa_signal_info signalInfo{};
+        if (0 == wpa_drv_signal_poll(wpa_s_0, &signalInfo))
+        {
+            info.mRssi    = signalInfo.current_signal; // dBm
+            info.mChannel = FrequencyToChannel(signalInfo.frequency);
+        }
+        else
+        {
+            // this values should be nullable according to the Matter spec
+            info.mRssi    = std::numeric_limits<decltype(info.mRssi)>::min();
+            info.mChannel = std::numeric_limits<decltype(info.mChannel)>::min();
+        }
+        return CHIP_NO_ERROR;
+    }
+
+    return CHIP_ERROR_INTERNAL;
+}
+
+uint8_t WiFiManager::GetSecurityType() const
+{
+    VerifyOrReturnValue(nullptr != mpWpaNetwork, EMBER_ZCL_SECURITY_TYPE_UNSPECIFIED);
+
+    if ((mpWpaNetwork->key_mgmt & WPA_KEY_MGMT_NONE) || !wpa_key_mgmt_wpa_any(mpWpaNetwork->key_mgmt))
+    {
+        return EMBER_ZCL_SECURITY_TYPE_NONE;
+    }
+    else if (wpa_key_mgmt_wpa_psk_no_sae(mpWpaNetwork->key_mgmt))
```

Comment:
I'm not able to review it with my current Wi-Fi knowledge, but I'll get back to it once I read more about it :)

Reply:
Great, thanks!